SeQuant/core/optimize.hpp

```diff
@@ -100,31 +100,49 @@ struct OptRes {
   eval_seq_t sequence;
 };
 
-/// returns a pair of index vectors
-/// first element of the pair is the vector of common indices compared by labels
-/// second element of the pair is the set symmetric difference of the input
-/// index vectors if either of the input index container is empty, the result is
-/// a pair of empty vectors
+///
+/// returns a vector of Index objects that are common in @c idxs1 and @c idxs2.
+///
 /// @note I1 and I2 containers are assumed to be sorted by using
 /// Index::LabelCompare{};
 template <typename I1, typename I2>
-std::pair<container::svector<Index>, container::svector<Index>> common_indices(
-    I1 const& idxs1, I2 const& idxs2) {
+container::svector<Index> common_indices(I1 const& idxs1, I2 const& idxs2) {
   using std::back_inserter;
   using std::begin;
   using std::end;
+  using std::set_intersection;
+
+  auto result = container::svector<Index>{};
+
+  set_intersection(begin(idxs1), end(idxs1), begin(idxs2), end(idxs2),
+                   back_inserter(result), Index::LabelCompare{});
+  return result;
+}
+
+///
+/// returns a vector of Index objects that are common in @c idxs1 and @c idxs2.
+///
+/// @note I1 and I2 containers are assumed to be sorted by using
```

Comment:
does this apply to `common_indices` also? if yes, update its docs. Complete docs ideally (copilot makes it easy)